src/SixLabors.Fonts/TextLayout.cs

```diff
@@ -14,48 +14,99 @@ namespace SixLabors.Fonts
     /// <summary>
     /// Encapsulated logic or laying out text.
     /// </summary>
-    internal class TextLayout
+    internal static class TextLayout
     {
-        internal static TextLayout Default { get; set; } = new();
-
-        public IReadOnlyList<GlyphLayout> GenerateLayout(ReadOnlySpan<char> text, TextOptions options)
+        public static IReadOnlyList<GlyphLayout> GenerateLayout(ReadOnlySpan<char> text, TextOptions options)
         {
             if (text.IsEmpty)
             {
                 return Array.Empty<GlyphLayout>();
             }
 
-            if (options.WrappingLength > 0)
+            TextBox textBox = ProcessText(text, options);
+            return LayoutText(textBox, options);
+        }
+
+        public static IReadOnlyList<TextRun> BuildTextRuns(ReadOnlySpan<char> text, TextOptions options)
+        {
+            if (options.TextRuns?.Count == 0)
```

Comment:
This doesn't work as expected if `options.TextRuns` is null. if `options.TextRuns` is null here then it still doesn't equal `0` as it equals `null` instead.